### PR TITLE
Remove repeat base64 decodes

### PR DESCRIPTION
**Pull request by GFJHogue**
Main issue: https://ucsc-cgl.atlassian.net/browse/SEAB-1680

Sibling PRs: https://github.com/dockstore/dockstore-deploy/pull/99 #19 

Consolidate base64 GitHub payload decoding in lambda

### CODE REVIEW

File: upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js [hunk 1]
```diff
@@ -102,21 +98,23 @@ function processEvent(event, callback) {
...
     var loneEvent = event.Records[0];
     var requestBody = JSON.parse(loneEvent.body);
-    if (! verifyGitHub(requestBody)) {
+
+    // The payload is URI-encoded and encoded in base64
+    const uriDecoded = decodeURIComponent(requestBody.payload);
+    const buff = Buffer.from(uriDecoded, 'base64');
+    const body = JSON.parse(buff.toString('utf8'));
+
+    console.log('GitHub Payload');
```
Comment (denis-yuen) [inline upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js:107]: thought: 
if we only really look at this when there is an unexpected problem, maybe this should be nested under the `if` on 110

**Comment by GFJHogue** [comment]
> Discussing this with @garyluu . I wonder if this is necessary and/or even harmful. Blocking for now.

@coverbeck Is this in reference to the payload log?

> Gary and I pasted this code into dev and we see the following error:
> 
> ```
> {
>     "errorType": "TypeError [ERR_INVALID_ARG_TYPE]",
>     "errorMessage": "The \"data\" argument must be one of type string, TypedArray, or DataView. Received type object",
>     "code": "ERR_INVALID_ARG_TYPE",
>     "stack": [
>         "TypeError [ERR_INVALID_ARG_TYPE]: The \"data\" argument must be one of type string, TypedArray, or DataView. Received type object",
>         "    at Hmac.update (internal/crypto/hash.js:58:11)",
>         "    at verifyGitHub (/var/task/index.js:18:66)",
>         "    at processEvent (/var/task/index.js:110:11)",
>         "    at Runtime.exports.handler (/var/task/index.js:217:5)",
>         "    at Runtime.handleOnce (/var/runtime/Runtime.js:66:25)"
>     ]
> }
> ```
> 
> I think you need to pass a string for the second arg on line 110.

D'oh, yes I missed that `verifyGitHub()` just uses the string and not the parsed JSON... will fix

**Comment by GFJHogue** [comment]
Updated title/description to reflect discussions here & in #19 - turns out we don't need to do URI-decoding.

File: upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js [hunk 2]
```diff
@@ -102,21 +98,22 @@ function processEvent(event, callback) {
...
     var loneEvent = event.Records[0];
     var requestBody = JSON.parse(loneEvent.body);
-    if (! verifyGitHub(requestBody)) {
+
+    // The payload is encoded in base64
+    const buff = Buffer.from(requestBody.payload, 'base64');
+    const bodyDecoded = buff.toString('utf8');
+    const body = JSON.parse(bodyDecoded);
+
+    if (! verifyGitHub(requestBody, bodyDecoded)) {
```
Comment (coverbeck) [inline upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js:107]: There is typically not a space after exclamation point, but we don't have a linter

**Pull request merged**